auth_from_http_remote_user/controllers/main.py

```diff
@@ -0,0 +1,62 @@
+# Author: Laurent Mignon
+# Copyright 2014-2018 'ACSONE SA/NV'
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl)
+
+import logging
+import werkzeug
+
+from odoo import http
+from odoo import api
+from odoo.http import request
+from odoo.addons.web.controllers import main
+
+from .. import utils
+
+_logger = logging.getLogger(__name__)
+
+
+class Home(main.Home):
+
+    _REMOTE_USER_ATTRIBUTE = 'HTTP_REMOTE_USER'
+
+    @http.route('/web', type='http', auth="none")
+    def web_client(self, s_action=None, **kw):
+        main.ensure_db()
+        try:
+            self._bind_http_remote_user(http.request.session.db)
+        except http.AuthenticationError:
+            return werkzeug.exceptions.Unauthorized().get_response()
+        return super(Home, self).web_client(s_action, **kw)
+
+    def _bind_http_remote_user(self, db_name):
+        headers = http.request.httprequest.headers.environ
+        login = headers.get(self._REMOTE_USER_ATTRIBUTE, None)
+        if not login:
+            # No SSO user in header, continue usual behavior
+            return
+        request_login = request.session.login
+        if request_login:
+            if request_login == login:
+                # Already authenticated
+                return
+            else:
+                request.session.logout(keep_db=True)
+        try:
+            user = utils.get_user(request.env['res.users'], login)
```

Comment:
any benefit in moving this to an util? Why not an instance method and `self.get_user`? It's easier to override it if needed too.

Reply:
I just did not think it belongs here and it would be needed to be overriden.
But if you think it is a bad idea I will change it